packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1563,9 +1603,27 @@ export class ContainerRuntime extends EventEmitter
                         || this.attachOpFiredForDataStore.has(key)),
                 )
                 .map(([key, value]) => {
-                    const snapshot = value.generateAttachMessage().snapshot;
-                    const componentSummary = convertToSummaryTree(snapshot, true);
-                    builder.addWithStats(key, componentSummary);
+                    if (value.isLoaded) {
+                        const snapshot = value.generateAttachMessage().snapshot;
+                        const treeWithStats = convertToSummaryTree(snapshot, true);
+                        builder.addWithStats(key, treeWithStats);
+                    } else {
+                        // If this data store is not yet loaded, then there should be no changes in the snapshot from
+                        // which it was created as it is detached container. So just use the previous snapshot.
+                        assert(this.context.baseSnapshot,
+                            "BaseSnapshot should be there as detached container loaded from snapshot");
+                        const summary: ISummarizeResult = {
+                            summary: convertSnapshotToSummaryTree(this.context.baseSnapshot.trees[key]),
+                            // No need to build stats, because we don't need them anyway here.
+                            stats: {
```

Comment:
Added stats. It is just that we mostly  use them for logging among other props. Here we don't log because we dont have much to log. So building stats for now maybe in future we may add some kind of telemetry event.